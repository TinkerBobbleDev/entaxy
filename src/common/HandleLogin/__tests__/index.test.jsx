import React from 'react'
import renderer from 'react-test-renderer'
import { HandleLoginComponent } from '../'

describe('HandleLogin', () => {
  const mochHandlePendingSignIn = jest.fn()

  it('matches snapshot with logged out user', () => {
    const component = renderer.create((
      <HandleLoginComponent
        handlePendingSignIn={jest.fn()}
      />
    ))
    expect(mochHandlePendingSignIn).not.toHaveBeenCalled()
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot with logged in user profile', () => {
    const component = renderer.create((
      <HandleLoginComponent
        user={{ isLoginPending: true }}
        handlePendingSignIn={mochHandlePendingSignIn}
      />
    ))
    expect(mochHandlePendingSignIn).toHaveBeenCalled()
    expect(component.toJSON()).toMatchSnapshot()
  })
})
