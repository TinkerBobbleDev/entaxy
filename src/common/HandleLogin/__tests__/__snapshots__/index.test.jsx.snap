// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandleLogin matches snapshot with logged in user profile 1`] = `
<div
  className="Login"
>
  <h2>
    Logging In...
  </h2>
</div>
`;

exports[`HandleLogin matches snapshot with logged out user 1`] = `null`;
